src/dotnet/commands/dotnet-restore3/Restore3CommandApp.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.CommandLine;
+
+namespace Microsoft.DotNet.Tools.Restore3
+{
+    public class Restore3CommandApp
+    {
+        public static int Execute(string[] args)
+        {
+            CommandLineApplication cmd = new CommandLineApplication
+            {
+                Name = "restore3",
+                FullName = "restore3",
+                Description = "restore for msbuild"
+            };
+
+            cmd.HelpOption("-h|--help");
+
+            var argRoot = cmd.Argument(
+                    "[root]",
+                    "Optional path to a project file or MSBuild arguments.",
+                    multipleValues: true);
+
+            var sourceOption = cmd.Option(
```

Comment:
> FallbackSource has been removed, I know it's used some places in the CLI build still. Is there a good reason to keep this instead of just having the user specify all sources with -s?

According to https://github.com/NuGet/Home/issues/2477, `-s` will completely override what is in NuGet.config.  This seems limiting since in a lot of cases you only want to add a source for this specific invocation.  Not completely overriding NuGet.config.  But there are other cases where you DO want to override.

Will both scenarios be supported by `-s`?